torch/nn/modules/activation.py

```diff
@@ -741,19 +744,20 @@ def _reset_parameters(self):
     @weak_script_method
     def forward(self, query, key, value, key_padding_mask=None, incremental_state=None,
                 need_weights=True, static_kv=False, attn_mask=None):
-        """
-        Inputs of forward function
-            query: [target length, batch size, embed dim]
-            key: [sequence length, batch size, embed dim]
-            value: [sequence length, batch size, embed dim]
-            key_padding_mask: if True, mask padding based on batch size
-            incremental_state: if provided, previous time steps are cashed
-            need_weights: output attn_output_weights
-            static_kv: key and value are static
-
-        Outputs of forward function
-            attn_output: [target length, batch size, embed dim]
-            attn_output_weights: [batch size, target length, sequence length]
+        r"""
+    Inputs
+        query: [target length, batch size, embed dim].
+        key: [sequence length, batch size, embed dim].
+        value: [sequence length, batch size, embed dim].
+        key_padding_mask: if True, mask padding based on batch size.
+        incremental_state: if provided, previous time steps are cashed.
```

Comment:
```suggestion
        incremental_state: if provided, previous time steps are cached.
```

Reply:
Also what does this actually accomplish and why is it important.

Reply:
I think Incremental state is useful at decoding time, i.e. when we are generating output incrementally since we don't want to do recompute most of the operations from the last step.
Static_kv is also somehow related to that.

But @ zhangguanheng66 or someone else can answer these questions in more detail.

Reply:
Thanks @tshrjn. incremental_state is just a dictionary used for storing previous states. See https://github.com/pytorch/fairseq/issues/166 for more details.

Reply:
I would suggest adding that information about what the incremental state does (and why it is useful) to the documentation.

Reply:
The static_kv is default to false. Unless the users are familiar with this function, they will most likely not touch it. I want to keep it short as possible.

Reply:
Then add such a warning to the function (or remove the functionality entirely). Short documentation does not equal concise and clear documentation.